packages/manager/modules/pci/src/projects/project/private-registry/credentials/credentials.controller.js

```diff
@@ -0,0 +1,72 @@
+import get from 'lodash/get';
+
+export default class PrivateRegistryDeleteCtrl {
+  /* @ngInject */
+  constructor(
+    $stateParams,
+    $translate,
+    $window,
+    privateRegistryService,
+  ) {
+    this.projectId = $stateParams.projectId;
+    this.registryId = $stateParams.registryId;
+    this.registryName = $stateParams.registryName;
+    this.harborURL = $stateParams.harborURL;
+    this.confirmationRequired = $stateParams.confirmationRequired;
+    this.fromState = $stateParams.fromState;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.privateRegistryService = privateRegistryService;
+    this.confirmed = false;
+    this.registryId = $stateParams.registryId;
+    this.isLoading = false;
+  }
+
+  $onInit() {
+    if (!this.confirmationRequired) {
+      this.getRegistryDetails()
+        .then(() => this.confirmRegeneration());
+    }
+  }
+
+  getRegistryDetails() {
+    this.isLoading = true;
```

Comment:
All the `get` actions could be passed through the routing `resolve` that way you don't need to handle loading or error catching.